devito/function.py

```diff
@@ -1554,7 +1598,7 @@ def _dist_scatter(self, data=None):
         data = scattered
 
         # Pack (reordered) coordinates so that they can be sent out via an Alltoallv
-        coords = self.coordinates.data[self._dist_subfunc_mask]
+        coords = self.coordinates.data[self._dist_subfunc_scatter_mask]
         # Send out the sparse point coordinates
         _, scount, sdisp, rshape, rcount, rdisp = self._dist_subfunc_alltoall
         scattered = np.empty(shape=rshape, dtype=self.coordinates.dtype)
```

Comment:
`coords/data = scattered` that is incorrect. Owned data is `data` no matter to how many process it is sent.